src/AllSyntaxPlugin.ts

```diff
@@ -12,14 +14,32 @@ export const ALL_PLUGINS = [
   'functionSent',
   'objectRestSpread',
   'dynamicImport',
-  'decorators'
+  'decorators',
+  'typescript'
 ];
 
 export default function(babel: typeof Babel) {
   return {
     manipulateOptions(opts: BabelOptions, parserOpts: ParseOptions) {
+      let isTypeScript = TypeScriptExtensions.has(extname(opts.filename));
+
+      parserOpts.sourceType = 'module';
+      parserOpts.allowImportExportEverywhere = true;
+      parserOpts.allowReturnOutsideFunction = true;
+      parserOpts.allowSuperOutsideMethod = true;
+
+      if (!parserOpts.plugins) {
+        parserOpts.plugins = [];
+      }
+
       for (let plugin of ALL_PLUGINS) {
-        if (plugin !== 'flow' || !opts.filename.endsWith('.ts')) {
+        let usePlugin =
```

Comment:
bit nasty to read. what about:
```
const plugins = ALL_PLUGINS.filter(plugin => {
  if (plugin === 'flow' && isTypeScript) return false;
  if (plugin === 'typescript' && !isTypeScript) return false;
  return true;
}); 
parserOpts.plugins.concat(...plugins);
```
or something similar?

Reply:
Good suggestions! I've gone with a hybrid.